File: 0414-third-maximum-number/0414-third-maximum-number.py
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        uniqueSet = set(nums)
        uniqueList = sorted(list(uniqueSet))
        listLength = len(uniqueList)
        
        if listLength < 3:
          return uniqueList[listLength - 1]

        return uniqueList[listLength - 3]

File: 0414-third-maximum-number/0414-third-maximum-number.ts
function thirdMax(nums: number[]): number {
  const arr = [...new Set(nums.sort((a, b) => a - b))];
  return arr[arr.length - 3] || arr.length === 3
      ? arr[arr.length - 3]
      : arr[arr.length - 1];
};